### PR TITLE
Implement Pause screen

### PR DESCRIPTION
**Pull request by TomasMorton**
- Button to pause
- Pause Update functions
- UI with resume / exit

### CODE REVIEW

File: Assets/Prefabs/UI/EndGameScreen.prefab [hunk 1]
```diff
@@ -1965,7 +1965,7 @@ GameObject:
   m_Icon: {fileID: 0}
   m_NavMeshLayer: 0
   m_StaticEditorFlags: 0
-  m_IsActive: 1
+  m_IsActive: 0
```
Comment (TomasMorton) [inline Assets/Prefabs/UI/EndGameScreen.prefab:1968]: Oops... end game should not be active by default

File: Assets/Scripts/Managers/GameStateManager.cs [hunk 2]
```diff
@@ -15,19 +15,21 @@ public enum GameStates
         Pause,
         End,
     }
-    
-    
+
```
Comment (tessapower) [inline Assets/Scripts/Managers/GameStateManager.cs:18]: We should add this class to the `Managers` namespace

File: Assets/Scripts/Managers/GameStateManager.cs [hunk 3]
```diff
@@ -15,19 +15,21 @@ public enum GameStates
...
+
     [SerializeField] private GameStates _currentState;
```
Comment (tessapower) [inline Assets/Scripts/Managers/GameStateManager.cs:20]: The naming convention for Unity serialized fields is the same as private fields without an underscore, so `_currentState` -> `currentState` and `_previousState` to `previousState` :)

File: Assets/Scripts/UI/HudController.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
+using UnityEngine;
+
+public class HudController : MonoBehaviour
+{
+    [SerializeField] private GameObject _ui;
```
Comment (tessapower) [inline Assets/Scripts/UI/HudController.cs:5]:
```diff
- [SerializeField] private GameObject _ui;
+ [SerializeField] private GameObject ui;
```

File: Assets/Scripts/UI/PauseScreenController.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class PauseScreenController : MonoBehaviour
+{
+    [SerializeField] private GameObject _ui;
```
Comment (tessapower) [inline Assets/Scripts/UI/PauseScreenController.cs:6]:
```diff
- [SerializeField] private GameObject _ui;
+ [SerializeField] private GameObject ui;
```

**Review by tessapower — COMMENTED** [review]
Looks pretty good!

File: Assets/Scripts/Managers/GameStateManager.cs (hunk 3 above)
Comment (TomasMorton) [reply Assets/Scripts/Managers/GameStateManager.cs:20]: Does that include fields that shouldn't be set, only viewed for debug purposes?

File: Assets/Scripts/UI/HudController.cs (hunk 4 above)
Comment (TomasMorton) [reply Assets/Scripts/UI/HudController.cs:5]: Fixed

File: Assets/Scripts/UI/PauseScreenController.cs (hunk 5 above)
Comment (TomasMorton) [reply Assets/Scripts/UI/PauseScreenController.cs:6]: Fixed

**Pull request closed without merging**